src/node/manifest/manifest.c

```diff
@@ -29,17 +45,10 @@ result_t generate_node_manifest(char *public_key_string, char *public_host,
   manifest.version = version;
 
   char node_id_string[512];
-  result_t res_public_key_hash =
-      openssl_hash_data(stripped_public_key, strlen(stripped_public_key));
-  char *public_key_hash = PROPAGATE(res_public_key_hash);
-
-  strcpy(node_id_string, "SHOGN");
-  strcat(node_id_string, &public_key_hash[32]);
+  PROPAGATE(node_id_from_public_key(stripped_public_key, node_id_string));
```

Comment:
In the definition of PROPAGATE at `./lib/netlibc/include/netlibc/error.h` line 60:

```
// WARN: DO NOT use a function call as the parameter for PROPAGATE e.g
// PROPAGATE(my_function());
```